### ISSUE TITLE
在训练开始时发生Segmentation fault (core dumped) at the begining of training

### ISSUE THREAD

**Issue by nihuizhidao** [PaddlePaddle/PaddleDetection#677]
训练刚刚开始时刚刚过了第一个iteration，就发生了Segmentation fault：

> loading annotations into memory...
> Done (t=0.00s)
> creating index...
> index created!
> 2020-05-14 17:13:37,547-INFO: 24 samples in file custom_dataset/tiantian_v3_seg_coco/annotations/instance_val.json
> 2020-05-14 17:13:37,548-INFO: places would be ommited when DataLoader is not iterable
> W0514 17:13:38.423727  3942 device_context.cc:237] Please NOTE: device: 0, CUDA Capability: 61, Driver API Version: 10.2, Runtime API Version: 10.0
> W0514 17:13:38.428025  3942 device_context.cc:245] device: 0, cuDNN Version: 7.6.
> 2020-05-14 17:13:39,522-INFO: Found /home/dell/.cache/paddle/weights/mask_rcnn_r50_fpn_1x
> 2020-05-14 17:13:39,523-INFO: Loading parameters from /home/dell/.cache/paddle/weights/mask_rcnn_r50_fpn_1x...
> 2020-05-14 17:13:39,758-WARNING: /tmp/tmpj0vboqhy/mask_rcnn_r50_fpn_1x.pdparams not found, try to load model file saved with [ save_params, save_persistables, save_vars ]
> 2020-05-14 17:13:39,758-WARNING: /tmp/tmpj0vboqhy/mask_rcnn_r50_fpn_1x.pdparams not found, try to load model file saved with [ save_params, save_persistables, save_vars ]
> loading annotations into memory...
> Done (t=0.02s)
> creating index...
> index created!
> 2020-05-14 17:13:40,095-INFO: 99 samples in file custom_dataset/tiantian_v3_seg_coco/annotations/instance_train.json
> 2020-05-14 17:13:40,096-INFO: places would be ommited when DataLoader is not iterable
> I0514 17:13:40.195358  3942 parallel_executor.cc:440] The Program will be executed on CUDA using ParallelExecutor, 2 cards are used, so 2 programs are executed in parallel.
> W0514 17:13:42.282743  3942 fuse_all_reduce_op_pass.cc:74] Find all_reduce operators: 84. To make the speed faster, some all_reduce ops are fused during training, after fusion, the number of all_reduce ops is 84.
> I0514 17:13:42.291462  3942 build_strategy.cc:365] SeqOnlyAllReduceOps:0, num_trainers:1
> I0514 17:13:42.634323  3942 parallel_executor.cc:307] Inplace strategy is enabled, when build_strategy.enable_inplace = True
> I0514 17:13:42.718104  3942 parallel_executor.cc:375] Garbage collection strategy is enabled, when FLAGS_eager_delete_tensor_gb = 0
> 2020-05-14 17:13:43,547-INFO: iter: 0, lr: 0.000003, 'loss_cls': '2.054725', 'loss_bbox': '0.205467', 'loss_rpn_cls': '0.102608', 'loss_rpn_bbox': '0.031138', 'loss_mask': '1.888842', 'loss': '4.282780', time: 0.058, eta: 0:19:22
> W0514 17:13:44.734632  4295 init.cc:209] Warning: PaddlePaddle catches a failure signal, it may not work properly
> W0514 17:13:44.734692  4295 init.cc:211] You could check whether you killed PaddlePaddle thread/process accidentally or report the case to PaddlePaddle
> W0514 17:13:44.734707  4295 init.cc:214] The detail failure signal is:
> 
> W0514 17:13:44.734725  4295 init.cc:217] *** Aborted at 1589447624 (unix time) try "date -d @1589447624" if you are using GNU date ***
> W0514 17:13:44.737655  4295 init.cc:217] PC: @                0x0 (unknown)
> W0514 17:13:44.737893  4295 init.cc:217] *** SIGSEGV (@0x100000029) received by PID 3942 (TID 0x7f77177fe700) from PID 41; stack trace: ***
> W0514 17:13:44.742903  4295 init.cc:217]     @     0x7f7854fe0390 (unknown)
> W0514 17:13:44.749749  4295 init.cc:217]     @     0x7f780b51f688 paddle::memory::detail::MemoryBlock::Split()
> W0514 17:13:44.758744  4295 init.cc:217]     @     0x7f780b51dbb2 _ZN6paddle6memory6detail14BuddyAllocator12SplitToAllocESt23_Rb_tree_const_iteratorISt5tupleIJmmPvEEEm
> W0514 17:13:44.768972  4295 init.cc:217]     @     0x7f780b51e095 paddle::memory::detail::BuddyAllocator::Alloc()
> W0514 17:13:44.776578  4295 init.cc:217]     @     0x7f780b50da35 paddle::memory::legacy::Alloc<>()
> W0514 17:13:44.785235  4295 init.cc:217]     @     0x7f780b50e9d5 paddle::memory::allocation::NaiveBestFitAllocator::AllocateImpl()
> W0514 17:13:44.790421  4295 init.cc:217]     @     0x7f780b507d43 paddle::memory::allocation::AllocatorFacade::Alloc()
> W0514 17:13:44.797467  4295 init.cc:217]     @     0x7f780b507fde paddle::memory::allocation::AllocatorFacade::AllocShared()
> W0514 17:13:44.802448  4295 init.cc:217]     @     0x7f780b50775c paddle::memory::AllocShared()
> W0514 17:13:44.807770  4295 init.cc:217]     @     0x7f780b4f3bb2 paddle::framework::Tensor::mutable_data()
> W0514 17:13:44.817210  4295 init.cc:217]     @     0x7f780acbb400 paddle::operators::GenerateProposalLabelsKernel<>::Compute()
> W0514 17:13:44.822504  4295 init.cc:217]     @     0x7f780acbca93 _ZNSt17_Function_handlerIFvRKN6paddle9framework16ExecutionContextEEZNKS1_24OpKernelRegistrarFunctorINS0_8platform8CPUPlaceELb0ELm0EJNS0_9operators28GenerateProposalLabelsKernelIfEENSA_IdEEEEclEPKcSF_iEUlS4_E_E9_M_invokeERKSt9_Any_dataS4_
> W0514 17:13:44.830654  4295 init.cc:217]     @     0x7f780b4637e6 paddle::framework::OperatorWithKernel::RunImpl()
> W0514 17:13:44.842005  4295 init.cc:217]     @     0x7f780b463fb1 paddle::framework::OperatorWithKernel::RunImpl()
> W0514 17:13:44.848700  4295 init.cc:217]     @     0x7f780b45d100 paddle::framework::OperatorBase::Run()
> W0514 17:13:44.857833  4295 init.cc:217]     @     0x7f780b1e2376 paddle::framework::details::ComputationOpHandle::RunImpl()
> W0514 17:13:44.866263  4295 init.cc:217]     @     0x7f780b199531 paddle::framework::details::FastThreadedSSAGraphExecutor::RunOpSync()
> W0514 17:13:44.876678  4295 init.cc:217]     @     0x7f780b19829f paddle::framework::details::FastThreadedSSAGraphExecutor::RunOp()
> W0514 17:13:44.879704  4295 init.cc:217]     @     0x7f780b198564 _ZNSt17_Function_handlerIFvvESt17reference_wrapperISt12_Bind_simpleIFS1_ISt5_BindIFZN6paddle9framework7details28FastThreadedSSAGraphExecutor10RunOpAsyncEPSt13unordered_mapIPNS6_12OpHandleBaseESt6atomicIiESt4hashISA_ESt8equal_toISA_ESaISt4pairIKSA_SC_EEESA_RKSt10shared_ptrINS5_13BlockingQueueImEEEEUlvE_vEEEvEEEE9_M_invokeERKSt9_Any_data
> W0514 17:13:44.890637  4295 init.cc:217]     @     0x7f7808b6c983 std::_Function_handler<>::_M_invoke()
> W0514 17:13:44.901640  4295 init.cc:217]     @     0x7f78088fac37 std::__future_base::_State_base::_M_do_set()
> W0514 17:13:44.904276  4295 init.cc:217]     @     0x7f7854fdda99 __pthread_once_slow
> W0514 17:13:44.906966  4295 init.cc:217]     @     0x7f780b193a52 _ZNSt13__future_base11_Task_stateISt5_BindIFZN6paddle9framework7details28FastThreadedSSAGraphExecutor10RunOpAsyncEPSt13unordered_mapIPNS4_12OpHandleBaseESt6atomicIiESt4hashIS8_ESt8equal_toIS8_ESaISt4pairIKS8_SA_EEES8_RKSt10shared_ptrINS3_13BlockingQueueImEEEEUlvE_vEESaIiEFvvEE6_M_runEv
> W0514 17:13:44.917790  4295 init.cc:217]     @     0x7f78088fce64 _ZZN10ThreadPoolC1EmENKUlvE_clEv
> W0514 17:13:44.919467  4295 init.cc:217]     @     0x7f7836393421 execute_native_thread_routine_compat
> W0514 17:13:44.921916  4295 init.cc:217]     @     0x7f7854fd66ba start_thread
> W0514 17:13:44.924396  4295 init.cc:217]     @     0x7f7854d0c41d clone
> W0514 17:13:44.926872  4295 init.cc:217]     @                0x0 (unknown)
> Segmentation fault (core dumped)

系统信息：

> Linux DL 4.15.0-72-generic #81~16.04.1-Ubuntu SMP Tue Nov 26 16:34:21 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux

> GTX 1080Ti*2，都用上了

`batch size =2`，使用如下配置（部分）：

> architecture: MaskRCNN
> use_gpu: true
> max_iters: 20000
> snapshot_iter: 200
> log_smooth_window: 20
> save_dir: output
> pretrain_weights: https://paddlemodels.bj.bcebos.com/object_detection/mask_rcnn_r50_fpn_1x.tar
> metric: COCO
> weights: output/mask_rcnn_r50_fpn_1x_tian/model_final
> num_classes: 7
> finetune_exclude_pretrained_params: ['cls_score', 'bbox_pred', 'mask_fcn_logits']

请问这个是什么原因呢？按照 #632 和 #619 的做法尝试了，但是还是出错。

**Comment by heavengate** [comment]
看报错在显存分配里面，是不是机器显存不够呢，可以确认一下GPU上是否有其他任务，剩余多少显存，或者把`batch_size=1`看下呢~

**Comment by qingqing01** [comment]
@nihuizhidao 看你是用了2个卡训练， GPU个数改变,batch-size变了之后，参考这里 https://github.com/PaddlePaddle/PaddleDetection/blob/release/0.2/docs/tutorials/FAQ.md#faq  调下参数，看下。

**Comment by nihuizhidao** [comment]
> 看报错在显存分配里面，是不是机器显存不够呢，可以确认一下GPU上是否有其他任务，剩余多少显存，或者把`batch_size=1`看下呢~

`batch_size=1`时，训练了几个iteration后也是出错了：

> loading annotations into memory...
> Done (t=0.00s)
> creating index...
> index created!
> 2020-05-14 18:44:14,783-INFO: 24 samples in file custom_dataset/tiantian_v3_seg_coco/annotations/instance_val.json
> 2020-05-14 18:44:14,784-INFO: places would be ommited when DataLoader is not iterable
> W0514 18:44:15.645292 22076 device_context.cc:237] Please NOTE: device: 0, CUDA Capability: 61, Driver API Version: 10.2, Runtime API Version: 10.0
> W0514 18:44:15.649588 22076 device_context.cc:245] device: 0, cuDNN Version: 7.6.
> 2020-05-14 18:44:16,734-INFO: Found /home/dell/.cache/paddle/weights/mask_rcnn_r50_fpn_1x
> 2020-05-14 18:44:16,734-INFO: Loading parameters from /home/dell/.cache/paddle/weights/mask_rcnn_r50_fpn_1x...
> 2020-05-14 18:44:16,970-WARNING: /tmp/tmpp8tvn14k/mask_rcnn_r50_fpn_1x.pdparams not found, try to load model file saved with [ save_params, save_persistables, save_vars ]
> 2020-05-14 18:44:16,970-WARNING: /tmp/tmpp8tvn14k/mask_rcnn_r50_fpn_1x.pdparams not found, try to load model file saved with [ save_params, save_persistables, save_vars ]
> loading annotations into memory...
> Done (t=0.02s)
> creating index...
> index created!
> 2020-05-14 18:44:17,307-INFO: 99 samples in file custom_dataset/tiantian_v3_seg_coco/annotations/instance_train.json
> 2020-05-14 18:44:17,308-INFO: places would be ommited when DataLoader is not iterable
> I0514 18:44:17.387063 22076 parallel_executor.cc:440] The Program will be executed on CUDA using ParallelExecutor, 2 cards are used, so 2 programs are executed in parallel.
> W0514 18:44:19.453794 22076 fuse_all_reduce_op_pass.cc:74] Find all_reduce operators: 84. To make the speed faster, some all_reduce ops are fused during training, after fusion, the number of all_reduce ops is 84.
> I0514 18:44:19.464352 22076 build_strategy.cc:365] SeqOnlyAllReduceOps:0, num_trainers:1
> I0514 18:44:19.811911 22076 parallel_executor.cc:307] Inplace strategy is enabled, when build_strategy.enable_inplace = True
> I0514 18:44:19.894541 22076 parallel_executor.cc:375] Garbage collection strategy is enabled, when FLAGS_eager_delete_tensor_gb = 0
> 2020-05-14 18:44:20,529-INFO: iter: 0, lr: 0.000003, 'loss_cls': '1.957837', 'loss_bbox': '0.218073', 'loss_rpn_cls': '0.054038', 'loss_rpn_bbox': '0.020294', 'loss_mask': '2.576046', 'loss': '4.826288', time: 0.038, eta: 0:12:41
> 2020-05-14 18:44:28,093-INFO: iter: 20, lr: 0.000006, 'loss_cls': '1.906165', 'loss_bbox': '0.193886', 'loss_rpn_cls': '0.028494', 'loss_rpn_bbox': '0.011841', 'loss_mask': '2.429716', 'loss': '4.624571', time: 0.519, eta: 2:52:47
> 2020-05-14 18:44:35,599-INFO: iter: 40, lr: 0.000008, 'loss_cls': '1.781431', 'loss_bbox': '0.168110', 'loss_rpn_cls': '0.038091', 'loss_rpn_bbox': '0.011454', 'loss_mask': '2.203896', 'loss': '4.151742', time: 0.376, eta: 2:05:07
> 2020-05-14 18:44:43,144-INFO: iter: 60, lr: 0.000010, 'loss_cls': '1.552254', 'loss_bbox': '0.210279', 'loss_rpn_cls': '0.029954', 'loss_rpn_bbox': '0.012544', 'loss_mask': '2.269466', 'loss': '4.040942', time: 0.374, eta: 2:04:18
> 2020-05-14 18:44:50,570-INFO: iter: 80, lr: 0.000010, 'loss_cls': '1.327620', 'loss_bbox': '0.161015', 'loss_rpn_cls': '0.032011', 'loss_rpn_bbox': '0.011324', 'loss_mask': '1.790294', 'loss': '3.372503', time: 0.373, eta: 2:03:46
> /home/dell/anaconda3/envs/paddle/lib/python3.7/site-packages/paddle/fluid/executor.py:782: UserWarning: The following exception is not an EOF exception.
>   "The following exception is not an EOF exception.")
> Traceback (most recent call last):
>   File "tools/train.py", line 326, in <module>
>     main()
>   File "tools/train.py", line 236, in main
>     outs = exe.run(compiled_train_prog, fetch_list=train_values)
>   File "/home/dell/anaconda3/envs/paddle/lib/python3.7/site-packages/paddle/fluid/executor.py", line 783, in run
>     six.reraise(*sys.exc_info())
>   File "/home/dell/anaconda3/envs/paddle/lib/python3.7/site-packages/six.py", line 703, in reraise
>     raise value
>   File "/home/dell/anaconda3/envs/paddle/lib/python3.7/site-packages/paddle/fluid/executor.py", line 778, in run
>     use_program_cache=use_program_cache)
>   File "/home/dell/anaconda3/envs/paddle/lib/python3.7/site-packages/paddle/fluid/executor.py", line 843, in _run_impl
>     return_numpy=return_numpy)
>   File "/home/dell/anaconda3/envs/paddle/lib/python3.7/site-packages/paddle/fluid/executor.py", line 677, in _run_parallel
>     tensors = exe.run(fetch_var_names)._move_to_list()
> paddle.fluid.core_avx.EnforceNotMet: 
> 
> --------------------------------------------
> C++ Call Stacks (More useful to developers):
> --------------------------------------------
> 0   std::string paddle::platform::GetTraceBackString<std::string const&>(std::string const&, char const*, int)
> 1   paddle::platform::EnforceNotMet::EnforceNotMet(std::string const&, char const*, int)
> 2   paddle::memory::detail::MemoryBlock::Split(paddle::memory::detail::MetadataCache*, unsigned long)
> 3   paddle::memory::detail::BuddyAllocator::SplitToAlloc(std::_Rb_tree_const_iterator<std::tuple<unsigned long, unsigned long, void*> >, unsigned long)
> 4   paddle::memory::detail::BuddyAllocator::Alloc(unsigned long)
> 5   void* paddle::memory::legacy::Alloc<paddle::platform::CPUPlace>(paddle::platform::CPUPlace const&, unsigned long)
> 6   paddle::memory::allocation::NaiveBestFitAllocator::AllocateImpl(unsigned long)
> 7   paddle::memory::allocation::AllocatorFacade::Alloc(paddle::platform::Place const&, unsigned long)
> 8   paddle::memory::allocation::AllocatorFacade::AllocShared(paddle::platform::Place const&, unsigned long)
> 9   paddle::memory::AllocShared(paddle::platform::Place const&, unsigned long)
> 10  paddle::framework::Tensor::mutable_data(paddle::platform::Place const&, paddle::framework::proto::VarType_Type, unsigned long)
> 11  std::vector<paddle::framework::Tensor, std::allocator<paddle::framework::Tensor> > paddle::operators::SampleRpnFgBgGt<float>(paddle::platform::CPUDeviceContext const&, paddle::framework::Tensor const&, int, float, float, float, std::linear_congruential_engine<unsigned long, 48271ul, 0ul, 2147483647ul>, bool)
> 12  paddle::operators::RpnTargetAssignKernel<float>::Compute(paddle::framework::ExecutionContext const&) const
> 13  std::_Function_handler<void (paddle::framework::ExecutionContext const&), paddle::framework::OpKernelRegistrarFunctor<paddle::platform::CPUPlace, false, 0ul, paddle::operators::RpnTargetAssignKernel<float>, paddle::operators::RpnTargetAssignKernel<double> >::operator()(char const*, char const*, int) const::{lambda(paddle::framework::ExecutionContext const&)#1}>::_M_invoke(std::_Any_data const&, paddle::framework::ExecutionContext const&)
> 14  paddle::framework::OperatorWithKernel::RunImpl(paddle::framework::Scope const&, paddle::platform::Place const&, paddle::framework::RuntimeContext*) const
> 15  paddle::framework::OperatorWithKernel::RunImpl(paddle::framework::Scope const&, paddle::platform::Place const&) const
> 16  paddle::framework::OperatorBase::Run(paddle::framework::Scope const&, paddle::platform::Place const&)
> 17  paddle::framework::details::ComputationOpHandle::RunImpl()
> 18  paddle::framework::details::FastThreadedSSAGraphExecutor::RunOpSync(paddle::framework::details::OpHandleBase*)
> 19  paddle::framework::details::FastThreadedSSAGraphExecutor::RunOp(paddle::framework::details::OpHandleBase*, std::shared_ptr<paddle::framework::BlockingQueue<unsigned long> > const&, unsigned long*)
> 20  std::_Function_handler<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Deleter> (), std::__future_base::_Task_setter<std::unique_ptr<std::__future_base::_Result<void>, std::__future_base::_Result_base::_Deleter>, void> >::_M_invoke(std::_Any_data const&)
> 21  std::__future_base::_State_base::_M_do_set(std::function<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Deleter> ()>&, bool&)
> 22  ThreadPool::ThreadPool(unsigned long)::{lambda()#1}::operator()() const
> 
> ------------------------------------------
> Python Call Stacks (More useful to users):
> ------------------------------------------
>   File "/home/dell/anaconda3/envs/paddle/lib/python3.7/site-packages/paddle/fluid/framework.py", line 2525, in append_op
>     attrs=kwargs.get("attrs", None))
>   File "/home/dell/anaconda3/envs/paddle/lib/python3.7/site-packages/paddle/fluid/layer_helper.py", line 43, in append_op
>     return self.main_program.current_block().append_op(*args, **kwargs)
>   File "/home/dell/anaconda3/envs/paddle/lib/python3.7/site-packages/paddle/fluid/layers/detection.py", line 419, in rpn_target_assign
>     'use_random': use_random
>   File "/home/dell/aas/schwarz/paddle_detection/PaddleDetection/ppdet/core/workspace.py", line 150, in partial_apply
>     return op(*args, **kwargs_)
>   File "/home/dell/aas/schwarz/paddle_detection/PaddleDetection/ppdet/modeling/anchor_heads/rpn_head.py", line 227, in get_loss
>     im_info=im_info)
>   File "/home/dell/aas/schwarz/paddle_detection/PaddleDetection/ppdet/modeling/architectures/mask_rcnn.py", line 108, in build
>     feed_vars['is_crowd'])
>   File "/home/dell/aas/schwarz/paddle_detection/PaddleDetection/ppdet/modeling/architectures/mask_rcnn.py", line 333, in train
>     return self.build(feed_vars, 'train')
>   File "tools/train.py", line 117, in main
>     train_fetches = model.train(feed_vars)
>   File "tools/train.py", line 326, in <module>
>     main()
> 
> ----------------------
> Error Message Summary:
> ----------------------
> Error: An error occurred here. There is no accurate error hint for this error yet. We are continuously in the process of increasing hint for this kind of error check. It would be helpful if you could inform us of how this conversion went by opening a github issue. And we will resolve it with high priority.
>   - New issue link: https://github.com/PaddlePaddle/Paddle/issues/new
>   - Recommended issue content: all error stack information
>   [Hint: Expected desc->total_size >= size, but received desc->total_size:0 < size:765952.] at (/paddle/paddle/fluid/memory/detail/memory_block.cc:41)
>   [operator < rpn_target_assign > error]
> terminate called without an active exception
> W0514 18:44:56.975338 22341 init.cc:209] Warning: PaddlePaddle catches a failure signal, it may not work properly
> W0514 18:44:56.975371 22341 init.cc:211] You could check whether you killed PaddlePaddle thread/process accidentally or report the case to PaddlePaddle
> W0514 18:44:56.975379 22341 init.cc:214] The detail failure signal is:
> 
> W0514 18:44:56.975389 22341 init.cc:217] *** Aborted at 1589453096 (unix time) try "date -d @1589453096" if you are using GNU date ***
> W0514 18:44:56.978327 22341 init.cc:217] PC: @                0x0 (unknown)
> W0514 18:44:56.978410 22341 init.cc:217] *** SIGABRT (@0x3e80000563c) received by PID 22076 (TID 0x7f7de75ea700) from PID 22076; stack trace: ***
> W0514 18:44:56.981022 22341 init.cc:217]     @     0x7f7e443d8390 (unknown)
> W0514 18:44:56.986177 22341 init.cc:217]     @     0x7f7e44032428 gsignal
> W0514 18:44:56.988729 22341 init.cc:217]     @     0x7f7e4403402a abort
> W0514 18:44:56.990798 22341 init.cc:217]     @     0x7f7e2777084a __gnu_cxx::__verbose_terminate_handler()
> W0514 18:44:56.993225 22341 init.cc:217]     @     0x7f7e2776ef47 __cxxabiv1::__terminate()
> W0514 18:44:56.995410 22341 init.cc:217]     @     0x7f7e2776ef7d std::terminate()
> W0514 18:44:56.997371 22341 init.cc:217]     @     0x7f7e2776ec5a __gxx_personality_v0
> W0514 18:44:57.001808 22341 init.cc:217]     @     0x7f7e431d2b97 _Unwind_ForcedUnwind_Phase2
> W0514 18:44:57.004391 22341 init.cc:217]     @     0x7f7e431d2e7d _Unwind_ForcedUnwind
> W0514 18:44:57.007002 22341 init.cc:217]     @     0x7f7e443d7070 __GI___pthread_unwind
> W0514 18:44:57.010177 22341 init.cc:217]     @     0x7f7e443cf845 __pthread_exit
> W0514 18:44:57.011445 22341 init.cc:217]     @     0x559e6b51e059 PyThread_exit_thread
> W0514 18:44:57.011842 22341 init.cc:217]     @     0x559e6b3a3c10 PyEval_RestoreThread.cold.799
> W0514 18:44:57.019235 22341 init.cc:217]     @     0x7f7df7c0f209 pybind11::gil_scoped_release::~gil_scoped_release()
> W0514 18:44:57.026257 22341 init.cc:217]     @     0x7f7df7bbbbc4 _ZZN8pybind1112cpp_function10initializeIZN6paddle6pybindL22pybind11_init_core_avxERNS_6moduleEEUlRNS2_9operators6reader22LoDTensorBlockingQueueERKSt6vectorINS2_9framework9LoDTensorESaISC_EEE60_bIS9_SG_EINS_4nameENS_9is_methodENS_7siblingEEEEvOT_PFT0_DpT1_EDpRKT2_ENUlRNS_6detail13function_callEE1_4_FUNESY_
> W0514 18:44:57.031545 22341 init.cc:217]     @     0x7f7df7c2d5b1 pybind11::cpp_function::dispatcher()
> W0514 18:44:57.032558 22341 init.cc:217]     @     0x559e6b4a4ab4 _PyMethodDef_RawFastCallKeywords
> W0514 18:44:57.033627 22341 init.cc:217]     @     0x559e6b4a4bd1 _PyCFunction_FastCallKeywords
> W0514 18:44:57.034474 22341 init.cc:217]     @     0x559e6b50ba39 _PyEval_EvalFrameDefault
> W0514 18:44:57.035259 22341 init.cc:217]     @     0x559e6b45066a _PyEval_EvalCodeWithName
> W0514 18:44:57.036039 22341 init.cc:217]     @     0x559e6b4514c5 _PyFunction_FastCallDict
> W0514 18:44:57.036880 22341 init.cc:217]     @     0x559e6b50854d _PyEval_EvalFrameDefault
> W0514 18:44:57.037602 22341 init.cc:217]     @     0x559e6b4a402b _PyFunction_FastCallKeywords
> W0514 18:44:57.038450 22341 init.cc:217]     @     0x559e6b506d40 _PyEval_EvalFrameDefault
> W0514 18:44:57.039185 22341 init.cc:217]     @     0x559e6b4a402b _PyFunction_FastCallKeywords
> W0514 18:44:57.040030 22341 init.cc:217]     @     0x559e6b506d40 _PyEval_EvalFrameDefault
> W0514 18:44:57.040827 22341 init.cc:217]     @     0x559e6b4513fb _PyFunction_FastCallDict
> W0514 18:44:57.041599 22341 init.cc:217]     @     0x559e6b470a73 _PyObject_Call_Prepend
> W0514 18:44:57.042454 22341 init.cc:217]     @     0x559e6b462fde PyObject_Call
> W0514 18:44:57.042773 22341 init.cc:217]     @     0x559e6b55fee7 t_bootstrap
> W0514 18:44:57.042901 22341 init.cc:217]     @     0x559e6b519968 pythread_wrapper
> W0514 18:44:57.045537 22341 init.cc:217]     @     0x7f7e443ce6ba start_thread
> Aborted (core dumped)

**Comment by nihuizhidao** [comment]
> @nihuizhidao 看你是用了2个卡训练， GPU个数改变,batch-size变了之后，参考这里 https://github.com/PaddlePaddle/PaddleDetection/blob/release/0.2/docs/tutorials/FAQ.md#faq 调下参数，看下。


我是使用了自己的数据集，做迁移学习，我的lr调小了很多，而且piecewise decay里的boundaries也是按照比例来的，如下：

> architecture: MaskRCNN
> use_gpu: true
> max_iters: 20000
> snapshot_iter: 200
> log_smooth_window: 20
> save_dir: output
> #pretrain_weights: https://paddle-imagenet-models-name.bj.bcebos.com/ResNet50_cos_pretrained.tar
> pretrain_weights: https://paddlemodels.bj.bcebos.com/object_detection/mask_rcnn_r50_fpn_1x.tar
> metric: COCO
> weights: output/mask_rcnn_r50_fpn_1x_tian/model_final
> num_classes: 7
> finetune_exclude_pretrained_params: ['cls_score', 'bbox_pred', 'mask_fcn_logits']

> LearningRate:
>   base_lr: 0.00001
>   schedulers:
>   - !PiecewiseDecay
>     gamma: 0.1
>     milestones: [13333, 17777]
>   - !LinearWarmup
>     start_factor: 0.3333333333333333
>     steps: 56

**Comment by jerrywgz** [comment]
从报错信息来看都是申请空间的时候报错，建议先使用fluid.layers.Print打印下每轮训练时的im_id，然后确认下报错时，对应图片的标注信息是否符合预期

**Comment by nihuizhidao** [comment]
这个跟 #1074 是一样的问题的，我close掉
